### PR TITLE
XM157 - Fix zeroing for secondary muzzles

### LINKED ISSUE #9647: Weapons with under barrel gl's can not adjust the zero with the ace_xm157_prototype  or any scope using the xm157 frame work attached
**Mods (complete and add to the following information):**
- **Arma 3:**  2.14 stable
- **CBA:**  CBA_A3 v3.16.1.231025
- **ACE3:**  3.16.1

**Description:**
Rifles with underbarrel grenade launcher you can not adjust the range (page up / page down) with the ace_xm157_prototype  or any scope using the xm157 frame work

**Steps to reproduce:**
_Follow [https://ace3.acemod.org/img/wiki/user/issue_flowchart.webp](this flowchart)!_

1. equip a rifle with a 203 or 420 grenade launcher, and a lot of 30mm rounds 
2. press page up or page down to change the zero , and see the range adjust,
3.  add the ace_xm157_prototype  with  `player addPrimaryWeaponItem "ace_xm157_prototype";`
4.  press page up or page down to change the zero , and see the range **not** adjust,

**Expected behavior:**
the gl zero should adjust with the ace_xm157_prototype  mounted the same as the gl zero adjusts without the ace_xm157_prototype  

**Where did the issue occur?**
- Have tested Dedicated, Self-Hosted Multiplayer, Singleplayer

**Log Files:**
can find no log file entries pertaining to the issue 

**Additional context:**
grenade launchers are fun 

**Screenshots:**

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title.
- Fixes #9647.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/xm157/functions/fnc_keyPress.sqf [hunk 1]
```diff
@@ -22,6 +22,11 @@ if (!GVAR(shown)) exitWith { false }; // fast exit if not shown
 if (!([ACE_player, objNull, ["isNotInside"]] call EFUNC(common,canInteractWith))) exitWith { false };
 if (!(ACE_player call CBA_fnc_canUseWeapon)) exitWith { false };
 
+private _scope = (ACE_player weaponAccessories currentMuzzle ACE_player) param [2, ""];
+
+if (_scope == "") exitWith { false }; // should not happen, but put as a precaution
+if (getText (_scope call CBA_fnc_getItemConfig >> "weaponInfoType") != "ace_xm157_info") exitWith { false }; // allow zeroing changes for muzzles not using a scope using the XM157 framework
```
Comment (veteran29) [inline addons/xm157/functions/fnc_keyPress.sqf:28]:
```suggestion
if (getText (_scope call CBA_fnc_getItemConfig >> "weaponInfoType") != QGVAR(info)) exitWith { false }; // allow zeroing changes for muzzles not using a scope using the XM157 framework
```

File: addons/xm157/functions/fnc_keyPress.sqf [hunk 2]
```diff
@@ -22,6 +22,11 @@ if (!GVAR(shown)) exitWith { false }; // fast exit if not shown
 if (!([ACE_player, objNull, ["isNotInside"]] call EFUNC(common,canInteractWith))) exitWith { false };
 if (!(ACE_player call CBA_fnc_canUseWeapon)) exitWith { false };
 
+private _scope = (ACE_player weaponAccessories currentMuzzle ACE_player) param [2, ""];
+
+if (_scope == "") exitWith { false }; // should not happen, but put as a precaution
+if (getText (_scope call CBA_fnc_getItemConfig >> "weaponInfoType") != QGVAR(info)) exitWith { false }; // allow zeroing changes for muzzles not using a scope using the XM157 framework
+
```
Comment (PabstMirror) [inline addons/xm157/functions/fnc_keyPress.sqf:29]: I don't think `weaponAccessories` supports muzzles so it's probably just easier to do
```suggestion
if (currentMuzzle ACE_player != currentWeapon ACE_player) exitWith { false };
```

Comment (johnb432) [reply addons/xm157/functions/fnc_keyPress.sqf:29]: > I don't think `weaponAccessories` supports muzzles so it's probably just easier to do

It reports `["", "", "", ""]` for the vanilla GL, so I assumed it works. But thinking about it a bit harder, it probably doesn't.

If we implement your suggestion, it means that non-primary muzzles cannot have XM157-framework based optics. Is that your intention?

Comment (PabstMirror) [reply addons/xm157/functions/fnc_keyPress.sqf:29]: I think that's fine
I don't think any GLs use the optic sights anyways
and if they could I don't think you have enough mil range in the sight (you need extreme angles to launch)
The only useful case might be that apex 5.56/50 cal combo rifle thing but that's pretty obscure

Comment (johnb432) [reply addons/xm157/functions/fnc_keyPress.sqf:29]: I'll check the combo rifle and see if it could a problem. If not, I'll commit your suggestion.

Comment (johnb432) [reply addons/xm157/functions/fnc_keyPress.sqf:29]: Combo rifle (classname `arifle_ARX_blk_F`) would be able to benefit from having the XM157 feature available for both muzzles. However, my proposed solution does not work.

But even more importantly:  The ballistics computer does not support secondary muzzles, so your solution is fine.

**Review by PabstMirror — APPROVED** [review]

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**